make pre-commit fails on Windows due to .venv/bin paths and newline mismatches

When running `make pre-commit` on Windows, the command fails for two main reasons:
1. **Makefile not Windows-compatible:** 
    The Makefile currently uses Unix-style paths (`.venv/bin/...`), but on Windows virtual environments use `Scripts/`.  
    This prevents running `make pre-commit` without manually editing the Makefile.
2. **Newline mismatch (CRLF vs LF):**  
    Some `.yaml` and `.txt` files fail `end-of-file-fixer` and `yamllint` hooks because they use CRLF instead of LF.  
    Example error:

```shell
- hook id: yamllint
- exit code: 1

rules_examples/start/advanced_start_rules.yaml
  1:7       error    wrong new line character: expected \n  (new-lines)
```

**Possible Solutions (?):**
1. **Update Makefile:**  
	Detect the OS and set correct paths for Python and pre-commit:

```make
ifeq ($(OS),Windows_NT)
PYTHON := .venv/Scripts/python.exe
PIP := .venv/Scripts/pip.exe
PRE_COMMIT := .venv/Scripts/pre-commit.exe
else
PYTHON := .venv/bin/python
PIP := .venv/bin/pip
PRE_COMMIT := .venv/bin/pre-commit
endif
```
2. **Ensure cross-platform newline handling:**
	Add a .gitattributes file to enforce LF endings and run `git add --renormalize .` to normalize tracked files:

```gitattributes
# .gitattributes
* text=auto eol=lf
```

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,0 +1,1 @@
+* text=auto

--- a/DEVELOPMENT_WORKFLOW.md
+++ b/DEVELOPMENT_WORKFLOW.md
@@ -13,8 +13,43 @@ cd core
 
 ### 2. Setup development environment
 
+Create a Python virtual environment (if you have not created it yet, If the virtual environment is already created, skip to the next command).
+
+```bash
+python -m venv .venv
+```
+
+and activate it before running the make commands.
+
+#### On Linux/macOS:
+```bash
+source .venv/bin/activate
+```
+#### On Windows:
+```cmd
+.venv\Scripts\activate
+```
+
+Run the setup command to install the required dependencies.
+
+**NOTE**: Use the `make` command if you installed make on Windows or you are on WSL. Use `dev-tools.ps1` or `dev-tools.bat` if you are on Windows and prefer not to install additional software.
+For more details, see [DEV_TOOLS.md](DEV_TOOLS.md).
+
+#### On Linux/macOS:
 ```bash
 make setup
+```
+
+#### On Windows:
+PowerShell:
+```powershell
+.\dev-tools.ps1 setup
+```
+or
+
+Command Prompt (Batch):
+```cmd
+.\dev-tools.bat setup
 ```
 
 This command:

--- a/DEVELOPMENT_WORKFLOW.md
+++ b/DEVELOPMENT_WORKFLOW.md
@@ -54,10 +54,12 @@ Command Prompt (Batch):
 
 This command:
 
-- Installs development dependencies from `requirements-dev.txt`
-- Configures pre-commit hooks for automatic code quality checking
+- Installs development dependencies from `requirements-dev.txt`.
+- Configures pre-commit hooks for automatic code quality checking.
 
 ### 3. Verify everything works
+
+Run the following command to check code formatting, linting, and tests before starting development. This ensures your environment is set up correctly and all pre-commit checks pass.
 
 ```bash
 make pre-commit
@@ -79,11 +81,15 @@ make install-dev
 
 #### Automatic code formatting
 
+Run the following command to automatically format your code according to the project's style guidelines.
+
 ```bash
 make format
 ```
 
 #### Code quality check
+
+Use this command to check your code for linting issues and ensure it meets quality standards.
 
 ```bash
 make lint
@@ -91,19 +97,15 @@ make lint
 
 #### Running tests
 
+Execute this command to run all tests and verify your changes do not break existing functionality.
+
 ```bash
 make test
 ```
 
-#### Complete check before commit
-
-```bash
-make pre-commit
-```
-
 ### 3. Before committing
 
-Pre-commit hooks run automatically, but you can run them manually:
+Pre-commit hooks run automatically, but you can run them manually.
 
 ```bash
 make pre-commit
@@ -202,7 +204,6 @@ mypy edge_mining/
 ### Formatting conflicts
 
 ```bash
-# If black and flake8 are in conflict:
 make format
 make lint
 

--- a/DEVELOPMENT_WORKFLOW.md
+++ b/DEVELOPMENT_WORKFLOW.md
@@ -32,7 +32,7 @@ source .venv/bin/activate
 
 Run the setup command to install the required dependencies.
 
-**NOTE**: Use the `make` command if you installed make on Windows or you are on WSL. Use `dev-tools.ps1` or `dev-tools.bat` if you are on Windows and prefer not to install additional software.
+**NOTE**: Use the `make` command if you are on Linux or you are on WSL. Use `dev-tools.ps1` or `dev-tools.bat` if you are on Windows.
 For more details, see [DEV_TOOLS.md](DEV_TOOLS.md).
 
 #### On Linux/macOS:

--- a/DEV_TOOLS.md
+++ b/DEV_TOOLS.md
@@ -3,11 +3,11 @@
 This project uses various tools to maintain code quality and includes a Makefile for common development tasks. The Makefile is cross-platform compatible and works on both Windows and Linux/macOS.
 
 ### Prerequisites for Windows users:
-To use the Makefile on Windows, you need to install `make`. You have several options:
-- Install [chocolatey](https://chocolatey.org/) and run: `choco install make`
+To use the Makefile on Windows, you must run `make` via Windows Subsystem for Linux (WSL).
 - Install [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install)
+- Run the `make` commands from a WSL shell (e.g., Ubuntu)
 
-**Alternative for Windows users:** If you prefer not to install make, you can use the provided scripts:
+**Alternative for Windows users:** If you prefer not to use WSL, you can use the provided alternative scripts:
 
 PowerShell:
 ```powershell

--- a/DEV_TOOLS.md
+++ b/DEV_TOOLS.md
@@ -1,6 +1,44 @@
 # Development Tools
 
-This project uses various tools to maintain code quality:
+This project uses various tools to maintain code quality and includes a Makefile for common development tasks. The Makefile is cross-platform compatible and works on both Windows and Linux/macOS.
+
+### Prerequisites for Windows users:
+To use the Makefile on Windows, you need to install `make`. You have several options:
+- Install [chocolatey](https://chocolatey.org/) and run: `choco install make`
+- Install [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install)
+
+**Alternative for Windows users:** If you prefer not to install make, you can use the provided scripts:
+
+PowerShell:
+```powershell
+.\dev-tools.ps1 help          # Show all available commands
+.\dev-tools.ps1 setup         # Set up development environment
+.\dev-tools.ps1 install       # Install dependencies
+# ... (same commands as make)
+```
+
+Command Prompt (Batch):
+```cmd
+.\dev-tools.bat help           # Show all available commands
+.\dev-tools.bat setup          # Set up development environment
+.\dev-tools.bat install        # Install dependencies
+# ... (same commands as make)
+```
+
+### Available commands:
+```bash
+make help          # Show all available commands
+make setup         # Set up development environment
+make install       # Install dependencies
+make install-dev   # Install development dependencies
+make format        # Format code with ruff
+make lint          # Run all linting checks
+make lint-fix      # Run linting and fix what can be auto-fixed
+make test          # Run tests
+make test-cov      # Run tests with coverage
+make pre-commit    # Run pre-commit hooks on all files
+make clean         # Clean cache and temporary files
+```
 
 ## Quick Setup
 

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-# Makefile per Edge Mining Development Tools
+# Makefile for Edge Mining Development Tools
 
 # Variables
 PYTHON := .venv/bin/python

--- a/Makefile
+++ b/Makefile
@@ -2,18 +2,25 @@
 
 # Detect operating system
 ifeq ($(OS),Windows_NT)
-    # Windows
-    VENV_BIN := .venv\Scripts
-    PYTHON := $(VENV_BIN)\python.exe
-    PIP := $(VENV_BIN)\pip.exe
-    PRE_COMMIT := $(VENV_BIN)\pre-commit.exe
+# On native Windows, instruct to use WSL or the provided scripts
+.PHONY: help setup install install-dev format lint lint-fix test test-cov pre-commit pre-commit-install clean
+help setup install install-dev format lint lint-fix test test-cov pre-commit pre-commit-install clean:
+	@echo "Windows environment detected."
+	@echo "This Makefile is intended to run under WSL (Linux) or on macOS/Linux."
+	@echo "Please either:"
+	@echo "  - Use WSL and run 'make <target>' from your Linux shell, or"
+	@echo "  - Use the Windows scripts: .\\dev-tools.ps1 or .\\dev-tools.bat"
+	@echo ""
+	@echo "Examples:"
+	@echo "  PowerShell: .\\dev-tools.ps1 help"
+	@echo "  CMD:        .\\dev-tools.bat help"
+	@exit
 else
-    # Unix-like (Linux, macOS)
-    VENV_BIN := .venv/bin
-    PYTHON := $(VENV_BIN)/python
-    PIP := $(VENV_BIN)/pip
-    PRE_COMMIT := $(VENV_BIN)/pre-commit
-endif
+	# Unix-like (Linux, macOS)
+	VENV_BIN := .venv/bin
+	PYTHON := $(VENV_BIN)/python
+	PIP := $(VENV_BIN)/pip
+	PRE_COMMIT := $(VENV_BIN)/pre-commit
 
 # Default target
 help:
@@ -99,3 +106,5 @@ clean:
 	find . -type d -name "*.egg-info" -exec rm -rf {} + 2>/dev/null || true
 	rm -rf build/ dist/ .coverage htmlcov/ .pytest_cache/ 2>/dev/null || true
 	@echo "✅ Cleanup complete!"
+
+endif

--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,19 @@
 # Makefile for Edge Mining Development Tools
 
-# Variables
-PYTHON := .venv/bin/python
-PIP := .venv/bin/pip
-PRE_COMMIT := .venv/bin/pre-commit
+# Detect operating system
+ifeq ($(OS),Windows_NT)
+    # Windows
+    VENV_BIN := .venv\Scripts
+    PYTHON := $(VENV_BIN)\python.exe
+    PIP := $(VENV_BIN)\pip.exe
+    PRE_COMMIT := $(VENV_BIN)\pre-commit.exe
+else
+    # Unix-like (Linux, macOS)
+    VENV_BIN := .venv/bin
+    PYTHON := $(VENV_BIN)/python
+    PIP := $(VENV_BIN)/pip
+    PRE_COMMIT := $(VENV_BIN)/pre-commit
+endif
 
 # Default target
 help:

--- a/README.md
+++ b/README.md
@@ -31,10 +31,17 @@ The project uses **Hexagonal Architecture (Ports and Adapters)** to clearly sepa
     cd core
     ```
 2.  **Create a virtual environment (recommended):**
+
+    #### On Linux/macOS:
     ```bash
     python -m venv .venv
-    source .venv/bin/activate  # On Linux/macOS
-    # .venv\Scripts\activate    # On Windows
+    source .venv/bin/activate
+    ```
+
+    #### On Windows:
+    ```cmd
+    python -m venv .venv
+    .venv\Scripts\activate
     ```
 3.  **Install dependencies:**
     ```bash

--- a/dev-tools.bat
+++ b/dev-tools.bat
@@ -1,0 +1,131 @@
+@echo off
+REM Batch script for Edge Mining Development Tools (Windows CMD alternative to Makefile)
+
+setlocal EnableDelayedExpansion
+
+REM Variables
+set VENV_PATH=.venv\Scripts
+set PYTHON=%VENV_PATH%\python.exe
+set PIP=%VENV_PATH%\pip.exe
+set PRE_COMMIT=%VENV_PATH%\pre-commit.exe
+
+REM Get command from first argument
+set COMMAND=%1
+if "%COMMAND%"=="" set COMMAND=help
+
+REM Main command dispatcher
+if /i "%COMMAND%"=="help" goto :help
+if /i "%COMMAND%"=="setup" goto :setup
+if /i "%COMMAND%"=="install" goto :install
+if /i "%COMMAND%"=="install-dev" goto :install-dev
+if /i "%COMMAND%"=="format" goto :format
+if /i "%COMMAND%"=="lint" goto :lint
+if /i "%COMMAND%"=="lint-fix" goto :lint-fix
+if /i "%COMMAND%"=="test" goto :test
+if /i "%COMMAND%"=="test-cov" goto :test-cov
+if /i "%COMMAND%"=="pre-commit" goto :pre-commit
+if /i "%COMMAND%"=="pre-commit-install" goto :pre-commit-install
+if /i "%COMMAND%"=="clean" goto :clean
+
+echo Unknown command: %COMMAND%
+echo.
+goto :help
+
+:help
+echo Edge Mining Development Tools (Batch)
+echo ====================================
+echo.
+echo Available commands:
+echo   setup          - Set up development environment
+echo   install        - Install dependencies
+echo   install-dev    - Install development dependencies
+echo   format         - Format code with ruff
+echo   lint           - Run all linting checks
+echo   lint-fix       - Run linting and fix what can be auto-fixed
+echo   test           - Run tests
+echo   test-cov       - Run tests with coverage
+echo   pre-commit     - Run pre-commit hooks on all files
+echo   pre-commit-install - Install pre-commit hooks
+echo   clean          - Clean cache and temporary files
+echo.
+echo Usage: dev-tools.bat ^<command^>
+echo Example: dev-tools.bat setup
+goto :end
+
+:setup
+call :install-dev
+call :pre-commit-install
+echo ✅ Development environment setup complete!
+goto :end
+
+:install
+echo 📦 Installing production dependencies...
+%PIP% install -r requirements.txt
+echo ✅ Production dependencies installed!
+goto :end
+
+:install-dev
+echo 📦 Installing development dependencies...
+%PIP% install -r requirements-dev.txt
+echo ✅ Development dependencies installed!
+goto :end
+
+:format
+echo 🔧 Formatting code...
+%PYTHON% -m ruff format edge_mining/ tests/
+echo ✅ Code formatting complete!
+goto :end
+
+:lint
+echo 🔍 Running linting checks...
+%PYTHON% -m ruff check edge_mining/
+%PYTHON% -m mypy edge_mining/
+%PYTHON% -m bandit -r edge_mining/ --skip B311,B104
+echo ✅ Linting complete!
+goto :end
+
+:lint-fix
+echo 🔧 Running auto-fixable linting...
+%PYTHON% -m ruff check --fix edge_mining/
+%PYTHON% -m ruff format edge_mining/
+echo ✅ Auto-fix complete!
+goto :end
+
+:test
+echo 🧪 Running tests...
+%PYTHON% -m pytest tests/ -v
+echo ✅ Tests complete!
+goto :end
+
+:test-cov
+echo 🧪 Running tests with coverage...
+%PYTHON% -m pytest tests/ -v --cov=edge_mining --cov-report=html --cov-report=term
+echo ✅ Tests with coverage complete!
+goto :end
+
+:pre-commit
+echo 🔧 Running pre-commit hooks...
+%PRE_COMMIT% run --all-files
+echo ✅ Pre-commit complete!
+goto :end
+
+:pre-commit-install
+echo 🔧 Installing pre-commit hooks...
+%PRE_COMMIT% install
+echo ✅ Pre-commit hooks installed!
+goto :end
+
+:clean
+echo 🧹 Cleaning cache and temporary files...
+for /d /r . %%d in (__pycache__) do @if exist "%%d" rd /s /q "%%d" 2>nul
+for /r . %%f in (*.pyc *.pyo) do @if exist "%%f" del /q "%%f" 2>nul
+for /d /r . %%d in (*.egg-info) do @if exist "%%d" rd /s /q "%%d" 2>nul
+if exist build rd /s /q build 2>nul
+if exist dist rd /s /q dist 2>nul
+if exist .coverage del /q .coverage 2>nul
+if exist htmlcov rd /s /q htmlcov 2>nul
+if exist .pytest_cache rd /s /q .pytest_cache 2>nul
+echo ✅ Cleanup complete!
+goto :end
+
+:end

--- a/dev-tools.ps1
+++ b/dev-tools.ps1
@@ -1,0 +1,144 @@
+# PowerShell script for Edge Mining Development Tools (Windows alternative to Makefile)
+
+param(
+    [Parameter(Position=0)]
+    [string]$Command = "help"
+)
+
+# Variables
+$VENV_PATH = ".venv\Scripts"
+$PYTHON = "$VENV_PATH\python.exe"
+$PIP = "$VENV_PATH\pip.exe"
+$PRE_COMMIT = "$VENV_PATH\pre-commit.exe"
+
+function Show-Help {
+    Write-Host "Edge Mining Development Tools (PowerShell)" -ForegroundColor Green
+    Write-Host "==========================================" -ForegroundColor Green
+    Write-Host ""
+    Write-Host "Available commands:" -ForegroundColor Yellow
+    Write-Host "  setup          - Set up development environment"
+    Write-Host "  install        - Install dependencies"
+    Write-Host "  install-dev    - Install development dependencies"
+    Write-Host "  format         - Format code with ruff"
+    Write-Host "  lint           - Run all linting checks"
+    Write-Host "  lint-fix       - Run linting and fix what can be auto-fixed"
+    Write-Host "  test           - Run tests"
+    Write-Host "  test-cov       - Run tests with coverage"
+    Write-Host "  pre-commit     - Run pre-commit hooks on all files"
+    Write-Host "  pre-commit-install - Install pre-commit hooks"
+    Write-Host "  clean          - Clean cache and temporary files"
+    Write-Host ""
+    Write-Host "Usage: .\dev-tools.ps1 <command>" -ForegroundColor Cyan
+    Write-Host "Example: .\dev-tools.ps1 setup" -ForegroundColor Cyan
+}
+
+function Install-Dependencies {
+    Write-Host "📦 Installing production dependencies..." -ForegroundColor Blue
+    & $PIP install -r requirements.txt
+    Write-Host "✅ Production dependencies installed!" -ForegroundColor Green
+}
+
+function Install-DevDependencies {
+    Write-Host "📦 Installing development dependencies..." -ForegroundColor Blue
+    & $PIP install -r requirements-dev.txt
+    Write-Host "✅ Development dependencies installed!" -ForegroundColor Green
+}
+
+function Setup-Environment {
+    Install-DevDependencies
+    Install-PreCommitHooks
+    Write-Host "✅ Development environment setup complete!" -ForegroundColor Green
+}
+
+function Format-Code {
+    Write-Host "🔧 Formatting code..." -ForegroundColor Blue
+    & $PYTHON -m ruff format edge_mining/ tests/
+    Write-Host "✅ Code formatting complete!" -ForegroundColor Green
+}
+
+function Run-Lint {
+    Write-Host "🔍 Running linting checks..." -ForegroundColor Blue
+    & $PYTHON -m ruff check edge_mining/
+    & $PYTHON -m mypy edge_mining/
+    & $PYTHON -m bandit -r edge_mining/ --skip B311,B104
+    Write-Host "✅ Linting complete!" -ForegroundColor Green
+}
+
+function Run-LintFix {
+    Write-Host "🔧 Running auto-fixable linting..." -ForegroundColor Blue
+    & $PYTHON -m ruff check --fix edge_mining/
+    & $PYTHON -m ruff format edge_mining/
+    Write-Host "✅ Auto-fix complete!" -ForegroundColor Green
+}
+
+function Run-Tests {
+    Write-Host "🧪 Running tests..." -ForegroundColor Blue
+    & $PYTHON -m pytest tests/ -v
+    Write-Host "✅ Tests complete!" -ForegroundColor Green
+}
+
+function Run-TestsWithCoverage {
+    Write-Host "🧪 Running tests with coverage..." -ForegroundColor Blue
+    & $PYTHON -m pytest tests/ -v --cov=edge_mining --cov-report=html --cov-report=term
+    Write-Host "✅ Tests with coverage complete!" -ForegroundColor Green
+}
+
+function Run-PreCommit {
+    Write-Host "🔧 Running pre-commit hooks..." -ForegroundColor Blue
+    & $PRE_COMMIT run --all-files
+    Write-Host "✅ Pre-commit complete!" -ForegroundColor Green
+}
+
+function Install-PreCommitHooks {
+    Write-Host "🔧 Installing pre-commit hooks..." -ForegroundColor Blue
+    & $PRE_COMMIT install
+    Write-Host "✅ Pre-commit hooks installed!" -ForegroundColor Green
+}
+
+function Clean-Cache {
+    Write-Host "🧹 Cleaning cache and temporary files..." -ForegroundColor Blue
+
+    # Remove __pycache__ directories
+    Get-ChildItem -Path . -Recurse -Directory -Name "__pycache__" | ForEach-Object {
+        Remove-Item -Path $_ -Recurse -Force -ErrorAction SilentlyContinue
+    }
+
+    # Remove .pyc and .pyo files
+    Get-ChildItem -Path . -Recurse -Include "*.pyc", "*.pyo" | Remove-Item -Force -ErrorAction SilentlyContinue
+
+    # Remove .egg-info directories
+    Get-ChildItem -Path . -Recurse -Directory -Name "*.egg-info" | ForEach-Object {
+        Remove-Item -Path $_ -Recurse -Force -ErrorAction SilentlyContinue
+    }
+
+    # Remove build and dist directories
+    @("build", "dist", ".coverage", "htmlcov", ".pytest_cache") | ForEach-Object {
+        if (Test-Path $_) {
+            Remove-Item -Path $_ -Recurse -Force -ErrorAction SilentlyContinue
+        }
+    }
+
+    Write-Host "✅ Cleanup complete!" -ForegroundColor Green
+}
+
+# Main command dispatcher
+switch ($Command.ToLower()) {
+    "help" { Show-Help }
+    "setup" { Setup-Environment }
+    "install" { Install-Dependencies }
+    "install-dev" { Install-DevDependencies }
+    "format" { Format-Code }
+    "lint" { Run-Lint }
+    "lint-fix" { Run-LintFix }
+    "test" { Run-Tests }
+    "test-cov" { Run-TestsWithCoverage }
+    "pre-commit" { Run-PreCommit }
+    "pre-commit-install" { Install-PreCommitHooks }
+    "clean" { Clean-Cache }
+    default {
+        Write-Host "Unknown command: $Command" -ForegroundColor Red
+        Write-Host ""
+        Show-Help
+        exit 1
+    }
+}

